src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/WPF/AppDotXamlDocument.cs

```diff
@@ -0,0 +1,944 @@
+                return;
+            }
+
+            // Replace just the string in the buffer with the new value.
+            Location replaceStart = DefinitionStart;
+            Location replaceEnd = DefinitionEndPlusOne;
+            string newText = EscapeXmlString(value);
+            if (_definitionIncludesQuotes)
+            {
+                newText = "\"" + newText + "\"";
+            }
+
+            // We know where to replace, so go ahead and do it.
+            replaceTextInstance.ReplaceText(replaceStart, replaceEnd, newText);
+        }
+    }
+
+    /// <summary>
+    /// Represents a property that was found in the XAML file in attribute syntax
+    /// </summary>
+    private class XamlPropertyInAttributeSyntax : XamlProperty
+    {
+        public XamlPropertyInAttributeSyntax(IVsTextLines vsTextLines, Location definitionStart, Location definitionEnd, string unescapedValue)
+            : base(vsTextLines, definitionStart, definitionEnd, unescapedValue, definitionIncludesQuotes: true)
+        {
+        }
+    }
+
+    /// <summary>
+    /// Represents a property that was found in property element syntax with a start and end tag.
+    /// </summary>
+    private class XamlPropertyInPropertyElementSyntax : XamlProperty
+    {
+        public XamlPropertyInPropertyElementSyntax(IVsTextLines vsTextLines, Location valueStart, Location valueEnd, string unescapedValue)
+            : base(vsTextLines, valueStart, valueEnd, unescapedValue, definitionIncludesQuotes: false)
+        {
+        }
+    }
+
+    /// <summary>
+    /// Represents a property that was found in property element syntax with an empty tag,
+    /// e.g. &lt;Application.StartupUri/>
+    /// </summary>
+    private class XamlPropertyInPropertyElementSyntaxWithEmptyTag : XamlPropertyInPropertyElementSyntax
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        private readonly string _fullyQualifiedPropertyName;
+
+        public XamlPropertyInPropertyElementSyntaxWithEmptyTag(IVsTextLines vsTextLines, string fullyQualifiedPropertyName, Location elementStart, Location elementEnd)
+            : base(vsTextLines, elementStart, elementEnd, unescapedValue: string.Empty)
+        {
+            _fullyQualifiedPropertyName = fullyQualifiedPropertyName;
+        }
+
+        public override string ActualDefinitionText => string.Empty;
+
+        public override void SetProperty(IReplaceText replaceTextInstance, string value)
+        {
+            if (UnescapedValue.Equals(value, StringComparison.Ordinal))
+            {
+                // The property value is not changing. Leave things alone.
+                return;
+            }
+
+            // Replace the empty tag in the buffer with a start/end element tag
+            // and the new value
+            string newText =
+$@"<{_fullyQualifiedPropertyName}>
+{EscapeXmlString(value)}
+</{_fullyQualifiedPropertyName}>";
```

Comment:
C# 10's raw strings can be used to keep things indented.

```suggestion
            string newText =
                $"""
                <{_fullyQualifiedPropertyName}>
                {EscapeXmlString(value)}
                </{_fullyQualifiedPropertyName}>
                """;
```